src/workerd/io/io-context.h

```diff
@@ -1283,10 +1283,8 @@ jsg::Promise<IoContext::MaybeIoOwn<addIoOwn, T>> IoContext::awaitIoImpl(
 
   // In the then() continuations below we grab a reference to the jsg::AsyncContextFrame
   // that is current when awaitIo is called and restore that frame when the promise is
-  // resolved. Why? The reasoning is straight forward: for promises, the context is captured
-  // when then is called, ensuring that when the continuation executes, it is running in that
-  // same context. When resolver.reject() or resolver.resolve() are called, v8 will pick up
-  // whatever the current context is and use it.
+  // resolved. This ensures that the promise continuations (including unhandled rejection
```

Comment:
I think this is incorrect now. This doesn't affect "promise continuations" (callbacks passed to `.then()`) -- they always inherit the context from where `.then()` was called. It only affects unhandled rejection events, which are not really continuations.

Reply:
I would say:

> The only reason we need to do this is so that "unhandledrejection" event handlers inherit the async context from where the I/O was initiated. If we didn't have that event, there'd be no need to capture the async context here.